addons/mrp/models/mrp_production.py

```diff
@@ -1523,9 +1523,9 @@ def _post_inventory(self, cancel_backorder=False):
         for order in self:
             finish_moves = order.move_finished_ids.filtered(lambda m: m.product_id == order.product_id and m.state not in ('done', 'cancel'))
             # the finish move can already be completed by the workorder.
-            if finish_moves and not finish_moves.quantity_done:
-                finish_moves._set_quantity_done(float_round(order.qty_producing - order.qty_produced, precision_rounding=order.product_uom_id.rounding, rounding_method='HALF-UP'))
-                finish_moves.move_line_ids.lot_id = order.lot_producing_id
+            for finish_move in finish_moves.filtered(lambda m: not m.quantity_done):
+                finish_move._set_quantity_done(float_round(order.qty_producing - order.qty_produced, precision_rounding=order.product_uom_id.rounding, rounding_method='HALF-UP'))
+                finish_move.move_line_ids.lot_id = order.lot_producing_id
```

Comment:
```suggestion
            for move in finish_moves:
                if move.quantity_done:
                    continue
                move._set_quantity_done(float_round(order.qty_producing - order.qty_produced, precision_rounding=order.product_uom_id.rounding, rounding_method='HALF-UP'))
                move.move_line_ids.lot_id = order.lot_producing_id
```

Reply:
@omra-odoo ok, apply my suggested change and I will merge

Reply:
@ticodoo changes have been done.

Thanks!